okd/src/use_okd_assets.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -eo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+MICROSHIFT_ROOT="${SCRIPTDIR}/../.."
+
+
+verify(){
+    okd_url=$1
+    okd_releaseTag=$2
+
+    #stdout=$(oc adm release info "${okd_url}:${okd_releaseTag}" 2>&1)
+    if ! stdout=$(oc adm release info "${okd_url}:${okd_releaseTag}" 2>&1)  ; then
+        echo -e "error verifying okd release (URL: ${okd_url} , TAG: ${okd_releaseTag}) \nERROR: ${stdout}"
+        exit 1
+    fi
+}
+
+replace_assets(){
+    okd_url=$1
+    okd_releaseTag=$2
+    arch=$(uname -m)
+
+    oc adm release info --image-for="${op}" "${okd_url}:${okd_releaseTag}"
+
+    # replace Microshift images with upstream (from OKD release)
+    for op in $(jq -e -r  '.images | keys []' "${MICROSHIFT_ROOT}/assets/release/release-${arch}.json") 
+    do
+        image=$(oc adm release info --image-for="${op}" "${okd_url}:${okd_releaseTag}" || true) 
+        if [ -n "${image}" ] ; then
+            echo "${op} ${image}"
+            jq --arg a "${op}" --arg b "${image}"  '.images[$a] = $b' "${MICROSHIFT_ROOT}/assets/release/release-${arch}.json" >"/tmp/release-${arch}.json.tmp"
+            mv "/tmp/release-${arch}.json.tmp" "${MICROSHIFT_ROOT}/assets/release/release-${arch}.json"
+        fi
+    done
+
+    pod_image=$(oc adm release info --image-for=pod "${okd_url}:${okd_releaseTag}" || true) 
+    # update the infra pods for crio
+    sed -i 's,pause_image .*,pause_image = '"\"${pod_image}\""',' packaging/crio.conf.d/10-microshift_amd64.conf
```

Comment:
This code needs to be updated to use $arch

Reply:
its trickier, i'll need to convert to it with `UNAME_TO_GOARCH_MAP=( ["x86_64"]="amd64" ["aarch64"]="arm64" )`

Reply:
Yes, please. We should not hardcode archs.